src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/pipeline_management.yml

```diff
@@ -0,0 +1,879 @@
+                Sid: "PassRole"
+                Action:
+                  - 'iam:PassRole'
+                Resource: '*'
+                Condition:
+                  StringEqualsIfExists:
+                    'iam:PassedToService':
+                      - cloudformation.amazonaws.com
+
+  ADFPipelineMangementCloudFormationRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - cloudformation.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: "adf-codepipeline-creation"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "codepipeline:CreatePipeline"
+                  - "codepipeline:DeletePipeline"
+                  - "codepipeline:DeleteWebhook"
+                  - "codepipeline:DeregisterWebhookWithThirdParty"
+                  - "codepipeline:GetPipeline"
+                  - "codepipeline:GetPipelineState"
+                  - "codepipeline:PutWebhook"
+                  - "codepipeline:RegisterWebhookWithThirdParty"
+                  - "codepipeline:StartPipelineExecution"
+                  - "codepipeline:TagResource"
+                  - "codepipeline:UpdatePipeline"
+                Resource:
+                  - !Sub arn:${AWS::Partition}:codepipeline:${AWS::Region}:${AWS::AccountId}:webhook:adf-webhook-*
+                  - !Sub arn:${AWS::Partition}:codepipeline:${AWS::Region}:${AWS::AccountId}:${PipelinePrefix}*
+              - Effect: Allow
+                Action:
+                  - "codebuild:CreateProject"
+                  - "codebuild:DeleteProject"
+                  - "codebuild:UpdateProject"
+                Resource:
+                  - !Sub arn:${AWS::Partition}:codebuild:${AWS::Region}:${AWS::AccountId}:project/adf-build-*
+              - Effect: Allow
+                Action:
+                  - "sns:DeleteTopic"
+                  - "sns:CreateTopic"
+                  - "sns:Unsubscribe"
+                  - "sns:Subscribe"
+                  - "sns:SetTopicAttributes"
+                  - "sns:GetTopicAttributes"
+                  - "sns:TagResource"
+                Resource:
+                  - !Sub arn:${AWS::Partition}:sns:${AWS::Region}:${AWS::AccountId}:${PipelinePrefix}*
+              - Effect: Allow
+                Action:
+                  - "iam:CreateRole"
+                  - "iam:CreateRolePolicy"
+                  - "iam:DeleteRole"
+                  - "iam:DeleteRolePolicy"
+                  - "iam:GetRole"
+                  - "iam:GetRolePolicy"
+                  - "iam:PassRole"
+                  - "iam:PutRolePolicy"
+                Resource:
+                  - !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:role/*
```

Comment:
I think this can be removed as well. Based on my reasoning this role should only be used to deploy CodePipeline stacks which don't include roles. 

Think this is just hangover from previous iterations.

Reply:
Unfortunately it does create a role depending on the resources in the pipeline.
For example, for CodePipeline it needs to create a role that has access.

That role name starts with `adf-pipeline-`. So maybe we can restrict this to only accept that as the prefix. What do you think?